### PR TITLE
Wallet screen with filler data

### PR DESCRIPTION
**Pull request by allenan**
Some of this is still pretty yolo, but first pass at the wallet screen with filler data.

I still want to:
- calculate the animation ranges dynamically based on device size
- explore making the activity card animation springier at the top/bottom ranges, and stickier in the center
- potentially driving the main animated value off of the flatlist scroll, depending on how we want that to work
- probably missed some typescript stuff here and there

And then in a future PR, I'll add data fetching

### CODE REVIEW

File: src/theme/theme.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ const palette = {
   purple: '#B377FF',
   purpleLight: '#6C71A3',
   purpleMain: '#474DFF',
+  purple300: '#343964',
```
Comment (allenan) [inline src/theme/theme.ts:25]: @matthewcarlreetz I was thinking we might want to move to a color naming scheme like this. Sort of like how ant.design does it (https://ant.design/docs/spec/colors#header). I think we'll end up in a painful place if we try to classify all the different color variants by dark, darker, darkest etc. It'd also maybe be good to use storybook or even something off-app to preview the different colors. I can maybe look into that.

**Comment by tyler-whitman** [comment]
This is so awesome! Not seeing anything alarming in the code haha.

Like you mentioned it would be super cool to animate the card up as your scroll down in the list to expand it. Looking so good already though!

File: src/features/wallet/root/WalletView.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+const WalletView = () => {
+  const { t } = useTranslation()
+
+  const handlePress = () => {
+    Haptic.trigger('notificationWarning')
+  }
+
+  const animatedValue = useRef(new Animated.Value(280)).current
+  animatedValue.addListener(({ value }) => console.log(value))
```
Comment (tyler-whitman) [inline src/features/wallet/root/WalletView.tsx:26]: Maybe remove this log if we dont need it

File: src/components/BarChart/Chart.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    onFocus(focusedBar)
+  }, [focusedBar, onFocus])
+
+  // SVG maths
+  const barWidth = width / (data.length + data.length - 1)
+  const maxUp = maxBy(data, 'up')?.up || 0
+  const maxDown = maxBy(data, 'down')?.down || 0
+  const maxBarHeight = maxUp + barWidth / 1.5 + maxDown
+  const vScale = (height - 20) / maxBarHeight
+  const minBarHeight = barWidth
```
Comment (tyler-whitman) [inline src/components/BarChart/Chart.tsx:39]: Is minBarHeight supposed to be barWidth?

**Review by tyler-whitman — APPROVED** [review]

File: src/components/BarChart/Chart.tsx (hunk 3 above)
Comment (allenan) [reply src/components/BarChart/Chart.tsx:39]: yeah it makes them round below that threshold

File: src/features/wallet/root/ActivityCard.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      isLast={index === data.length - 1}
+    />
+  )
+
+  const handlePress = () => {
+    Haptic.trigger('notificationWarning')
+  }
+
+  return (
+    <SlidingUpPanel
```
Comment (allenan) [inline src/features/wallet/root/ActivityCard.tsx:33]: FYI I think I'm going to end up removing this in favor of a custom pan responder in order to get the springy effect

File: src/theme/theme.ts (hunk 1 above)
Comment (matthewcarlreetz) [reply src/theme/theme.ts:25]: I TOTALLY agree :)

Comment (matthewcarlreetz) [reply src/theme/theme.ts:25]: @allenan I've been meaning to ask @PeterMain about getting a color palette. 

I'm most familiar with https://material.io/design/color/the-color-system.html#color-theme-creation, but I'm up for whatever.

**Review by matthewcarlreetz — COMMENTED** [review]
This is looking great! Most of my comments are regarding types and theming. As much as possible I want to rely on the theme so that if we do support dark/light mode it will be easier to support.

File: src/features/wallet/root/WalletView.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        paddingHorizontal="l"
+        backgroundColor="primaryBackground"
+        zIndex={1}
+      >
+        <Text variant="header" fontSize={22}>
+          {t('wallet.title')}
+        </Text>
+
+        <Box flexDirection="row" justifyContent="space-between" width={85}>
+          <TouchableOpacity onPress={handlePress}>
```
Comment (matthewcarlreetz) [inline src/features/wallet/root/WalletView.tsx:70]: Could use `src/components/TouchableOpacityBox.tsx` in place of these

File: src/components/BarChart/Chart.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+type Props = {
+  width: number
+  height: number
+  data: ChartData[]
+  onFocus: (data: ChartData | null) => void
+}
+
+const BarChart = ({ width, height, data, onFocus }: Props) => {
+  const [focusedBar, setFocusedBar] = useState(null)
```
[the remark this answers is not in the document]
Comment (allenan) [reply src/components/BarChart/Chart.tsx:23]: ahh that makes sense. I was trying to figure out how to make that error go away

File: src/components/BarChart/Chart.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react'
+import Svg, { Text, Rect } from 'react-native-svg'
+import { PanResponder, Animated } from 'react-native'
+import { maxBy, clamp, max } from 'lodash'
+import Haptic from 'react-native-haptic-feedback'
```
[the remark this answers is not in the document]
Comment (allenan) [reply src/components/BarChart/Chart.tsx:5]: it looks like that excludes android. can we include android in that?

[the remark this answers is not in the document]
Comment (allenan) [reply src/components/BarChart/Chart.tsx:5]: looks like it has the same haptic types so yeah i'll switch over to that

File: src/features/wallet/root/BalanceCard.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        flexDirection="row"
+        justifyContent="space-between"
+        alignItems="center"
+        style={[{ transform: [{ translateY }, { scale }] }]}
+      >
+        <Box>
+          <Text color="white" fontSize={40} fontWeight="300">
+            23,987
+          </Text>
+          <Text
```
[the remark this answers is not in the document]
Comment (allenan) [reply src/features/wallet/root/BalanceCard.tsx]: yeah i kind of got overwhelmed by trying to map all the variants in the design to specific variants in the theme. But let me see if these could be defined

File: src/components/BarChart/Chart.tsx (hunk 3 above)
Comment (tyler-whitman) [reply src/components/BarChart/Chart.tsx:39]: Ok just wanted to be sure

File: src/components/BarChart/Chart.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (tyler-whitman) [reply src/components/BarChart/Chart.tsx:5]: I remember Anthony doing something custom for Android haptics a while back? Was that something with the expo one?

[the remark this answers is not in the document]
Comment (allenan) [reply src/components/BarChart/Chart.tsx]: I think the custom thing was just excluding android from haptic haha. Let's see how it works with android included and we can always remap the impact style or just exclude altogether if it's bad.

File: src/features/wallet/root/ActivityItem.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import { TouchableWithoutFeedback } from 'react-native'
+import { formatDistanceToNow } from 'date-fns'
+import { round } from 'lodash'
+import animalHash from 'angry-purple-tiger'
```
[the remark this answers is not in the document]
Comment (allenan) [reply src/features/wallet/root/ActivityItem.tsx:5]: ah nice, I should update angry-purple-tiger at some point to export types

File: src/features/wallet/root/WalletView.tsx (hunk 5 above)
Comment (allenan) [reply src/features/wallet/root/WalletView.tsx]: sure can!

File: src/features/wallet/root/ActivityItem.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (matthewcarlreetz) [reply src/features/wallet/root/ActivityItem.tsx:5]: lol. I have no idea what it actually is, but it sure sounds cool.

**Review by matthewcarlreetz — APPROVED** [review]

**Pull request merged**